aten/src/ATen/native/mps/operations/BinaryKernel.mm

```diff
@@ -1,10 +1,387 @@
+      std::vector<uint32_t> iterShapeData(iterShape.size());
+      std::vector<std::array<uint32_t, nOffsets>> strides(nDim);
+
+      if (!allContiguous) {
+        for (const auto i: c10::irange(iterShape.size())) {
+          TORCH_CHECK(i <= UINT32_MAX);
+          iterShapeData[i] = (uint32_t)(iterShape[i]);
+        }
+
+        for (const auto i: c10::irange(nDim)) {
+          for (const auto offset: c10::irange(nOffsets)) {
+              auto final_offset = offset;
+              if (tensor_pos) {
+                if (offset == 1) {
+                  final_offset = tensor_pos;
+                } else if (offset == 2) {
+                  final_offset = scalar_pos;
+                }
+              }
+              strides[i][offset] = iter.strides(final_offset)[i];
+          }
+        }
+      }
+
+      std::string kernel = op;
+      kernel += "_kernel";
+      if (all_scalar) {
+        kernel += "_scalar";
+      }
+      if (scalar_pos) {
+        kernel += "_scalar_rhs";
+      }
+      if (!allContiguous) {
+        kernel += "_strided";
+      }
+
+      if (allContiguous) {
+        // Round up to next multiple of 4
+        numThreads = (numThreads + (4 - numThreads % 4)) / 4;
+        gridSize = MTLSizeMake(numThreads, 1, 1);
+      }
+
+      id<MTLComputePipelineState> binaryPSO = mps::getBinaryPSO(device,
+                                                          getMetalScalarType(outputTensor),
+                                                          getMetalScalarType(inputTensor),
+                                                          getMetalScalarType(otherTensor),
+                                                          getMetalScalarType(common_dtype),
+                                                          kernel,
+                                                          op,
+                                                          kernel_operator,
+                                                          type);
+      [computeEncoder setComputePipelineState:binaryPSO];
+      [computeEncoder setBuffer:inputBuffer  offset:inputTensorStorage atIndex:0];
+      [computeEncoder setBuffer:otherBuffer  offset:otherTensorStorage atIndex:1];
+      [computeEncoder setBuffer:outputBuffer offset:outputTensor.storage_offset() * outputTensor.element_size() atIndex:2];
+      if (!allContiguous) {
+        [computeEncoder setBytes:iterShapeData.data() length:sizeof(uint32_t) * iterShape.size() atIndex:3];
+        [computeEncoder setBytes:&nDim length:sizeof(uint32_t) atIndex:4];
+        [computeEncoder setBytes:strides.data() length:sizeof(uint32_t) * nDim * nOffsets atIndex:5];
+      }
+
+      NSUInteger tgSize = binaryPSO.maxTotalThreadsPerThreadgroup;
+      if (tgSize > numThreads) {
+          tgSize = numThreads;
+      }
+
+      MTLSize threadGroupSize = MTLSizeMake(tgSize, 1, 1);
+      [computeEncoder dispatchThreads: gridSize
+                threadsPerThreadgroup: threadGroupSize];
+    }
```

Comment:
If we want Profiler AND the Adaptive Commit for this kernel, you need to add this line here:
`mpsStream->commitAdaptive({inputTensor, otherTensor, outputTensor}, binaryPSO);`